src/state/stake/hooks.ts

```diff
@@ -742,6 +751,46 @@ function getSearchFiltered(info: any, search: string) {
   }
 }
 
+export function getStakingFees(
+  stakingInfo: StakingBasic | DualStakingBasic,
+  balanceState?: CallState,
+  totalSupplyState?: CallState,
+) {
+  let oneYearFeeAPY = 0;
+  let oneDayFee = 0;
+  let accountFee = 0;
+  if (pairs !== undefined) {
+    oneYearFeeAPY = pairs[stakingInfo.pair]?.oneDayVolumeUSD;
+
+    if (
+      oneYearFeeAPY &&
+      balanceState &&
+      balanceState.result &&
+      balanceState.result[0] &&
+      totalSupplyState &&
+      totalSupplyState.result &&
+      totalSupplyState.result[0]
+    ) {
+      const totalSupply = web3.utils.toWei(
+        pairs[stakingInfo.pair]?.totalSupply,
+        'ether',
+      );
+      const ratio =
+        Number(totalSupplyState.result[0].toString()) / Number(totalSupply);
```

Comment:
Number(totalSupplyState.result[0].toString()) why both toStrong and cast to number?

Also this is done twice put it in a variable and reuse it